determined-ai/determined
# feat: add create workspace namespace binding to webui

## Ticket


## Description

Added Namespace inputs to Create and Edit Workspace Modal


## Test Plan

Non Kubernetes RM:
1. Verify when you go to Create a Workspace or Edit a workspace from the UI, there is no place to create workspace namespace Bindings.

Kubernetes RM (single):
1. Go to the webui and Create a workspace. There should only be one place to input a namespace name. Create it without inputting anything there.
2. Try to edit the same workspace, it should pre-populate the same input with "default". Change it a valid namespace from that cluster. Save and it should save without errors. 
3. Create a workload in that workspace and verify it goes to the correct namespace.
4. Open the Edit workspace Modal again and verify it shows the updated namespace in that input box.
5. Create another workspace, and this time input a valid namespace during creation.
6. Verify it shows the updated namespace in the Edit Workspace Modal.

Multi RM:
1. Create a workspace and verify that it shows n number of input boxes for n number of Kubernetes RMs.
2. Input valid namespaces in all of them.
3. Verify it populates correctly in the Edit Workspace Modal.
4. Backspace one of these namespaces and save the workspace. Reopen the Edit Workspace modal and it should change to "default" or the default namespace configured for that cluster.


## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
